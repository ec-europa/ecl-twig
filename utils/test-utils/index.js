const Twig = require('twig');
const { format } = require('prettier');

const renderTwigFile = (file, options, cb) =>
  Twig.renderFile(file, options, (err, html) =>
    cb(err, format(html, { parser: 'html' }))
  );

const renderTwigFileAsNode = (file, options) =>
  new Promise((resolve, reject) =>
    Twig.renderFile(file, options, (err, html) => {
      if (err) return reject(err);

      const div = document.createElement('div');
      div.innerHTML = html.trim();

      return resolve(div.firstChild);
    })
  );

module.exports = {
  renderTwigFile,
  renderTwigFileAsNode,
};
