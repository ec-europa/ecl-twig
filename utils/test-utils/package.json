{
  "name": "@ecl-twig/test-utils",
  "version": "2.29.0",
  "description": "A local test helper",
  "main": "index.js",
  "private": true,
  "devDependencies": {
    "deepmerge": "4.2.2"
  }
}
