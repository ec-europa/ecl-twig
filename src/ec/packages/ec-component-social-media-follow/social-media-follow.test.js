import path from 'path';
import { renderTwigFileAsNode } from '@ecl-twig/test-utils';

import demoData from './demo/data';

demoData.links.forEach(link => {
  if (link.icon) {
    link.icon.forEach(icon => {
      icon.path = 'example'; // eslint-disable-line no-param-reassign
    });
  }
});

describe('EC - Social Media Follow', () => {
  const template = path.resolve(__dirname, './social-media-follow.html.twig');
  const render = params => renderTwigFileAsNode(template, params);

  describe('Horizontal', () => {
    test('renders correctly', () => {
      expect.assertions(1);

      return expect(render(demoData)).resolves.toMatchSnapshot();
    });
  });

  describe('Vertical', () => {
    test('renders correctly', () => {
      expect.assertions(1);

      return expect(
        render({
          ...demoData,
          variant: 'vertical',
        })
      ).resolves.toMatchSnapshot();
    });
  });
});
