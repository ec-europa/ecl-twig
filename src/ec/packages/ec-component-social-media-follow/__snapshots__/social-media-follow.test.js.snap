// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EC - Social Media Follow Horizontal renders correctly 1`] = `
"<div class=\\"ecl-social-media-follow\\">
  <p class=\\"ecl-social-media-follow__description\\">
    Follow the latest progress and learn more about getting involved.
  </p>
  <ul class=\\"ecl-social-media-follow__list\\">
    <li class=\\"ecl-social-media-follow__item\\">
      <a
        href=\\"/example\\"
        class=\\"ecl-social-media-follow__link ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before\\"
        ><svg
          focusable=\\"false\\"
          aria-hidden=\\"true\\"
          class=\\"ecl-link__icon ecl-icon ecl-icon--xl ecl-social-media-follow__icon\\"
        >
          <use xlink:href=\\"example#twitter\\" /></svg
        ><svg
          focusable=\\"false\\"
          aria-hidden=\\"true\\"
          class=\\"ecl-link__icon ecl-icon ecl-icon--xl ecl-social-media-follow__icon-hover\\"
        >
          <use xlink:href=\\"example#twitter_hover\\" /></svg
        ><span class=\\"ecl-link__label\\">Twitter</span></a
      >
    </li>
    <li class=\\"ecl-social-media-follow__item\\">
      <a
        href=\\"/example\\"
        class=\\"ecl-social-media-follow__link ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before\\"
        ><svg
          focusable=\\"false\\"
          aria-hidden=\\"true\\"
          class=\\"ecl-link__icon ecl-icon ecl-icon--xl ecl-social-media-follow__icon\\"
        >
          <use xlink:href=\\"example#facebook\\" /></svg
        ><svg
          focusable=\\"false\\"
          aria-hidden=\\"true\\"
          class=\\"ecl-link__icon ecl-icon ecl-icon--xl ecl-social-media-follow__icon-hover\\"
        >
          <use xlink:href=\\"example#facebook_hover\\" /></svg
        ><span class=\\"ecl-link__label\\">Facebook</span></a
      >
    </li>
    <li class=\\"ecl-social-media-follow__item\\">
      <a
        href=\\"/example\\"
        class=\\"ecl-social-media-follow__link ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before\\"
        ><svg
          focusable=\\"false\\"
          aria-hidden=\\"true\\"
          class=\\"ecl-link__icon ecl-icon ecl-icon--xl ecl-social-media-follow__icon\\"
        >
          <use xlink:href=\\"example#gplus\\" /></svg
        ><svg
          focusable=\\"false\\"
          aria-hidden=\\"true\\"
          class=\\"ecl-link__icon ecl-icon ecl-icon--xl ecl-social-media-follow__icon-hover\\"
        >
          <use xlink:href=\\"example#gplus_hover\\" /></svg
        ><span class=\\"ecl-link__label\\">Google+</span></a
      >
    </li>
    <li class=\\"ecl-social-media-follow__item\\">
      <a
        href=\\"/example\\"
        class=\\"ecl-social-media-follow__link ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before\\"
        ><svg
          focusable=\\"false\\"
          aria-hidden=\\"true\\"
          class=\\"ecl-link__icon ecl-icon ecl-icon--xl ecl-social-media-follow__icon\\"
        >
          <use xlink:href=\\"example#linkedin\\" /></svg
        ><svg
          focusable=\\"false\\"
          aria-hidden=\\"true\\"
          class=\\"ecl-link__icon ecl-icon ecl-icon--xl ecl-social-media-follow__icon-hover\\"
        >
          <use xlink:href=\\"example#linkedin_hover\\" /></svg
        ><span class=\\"ecl-link__label\\">Linkedin</span></a
      >
    </li>
    <li class=\\"ecl-social-media-follow__item\\">
      <a
        href=\\"/example\\"
        class=\\"ecl-social-media-follow__link ecl-link ecl-link--standalone\\"
        ><span class=\\"ecl-link__label\\">Other social networks</span></a
      >
    </li>
  </ul>
</div>
"
`;

exports[`EC - Social Media Follow Vertical renders correctly 1`] = `
"<div class=\\"ecl-social-media-follow ecl-social-media-follow--vertical\\">
  <p class=\\"ecl-social-media-follow__description\\">
    Follow the latest progress and learn more about getting involved.
  </p>
  <ul class=\\"ecl-social-media-follow__list\\">
    <li class=\\"ecl-social-media-follow__item\\">
      <a
        href=\\"/example\\"
        class=\\"ecl-social-media-follow__link ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before\\"
        ><svg
          focusable=\\"false\\"
          aria-hidden=\\"true\\"
          class=\\"ecl-link__icon ecl-icon ecl-icon--xl ecl-social-media-follow__icon\\"
        >
          <use xlink:href=\\"example#twitter\\" /></svg
        ><svg
          focusable=\\"false\\"
          aria-hidden=\\"true\\"
          class=\\"ecl-link__icon ecl-icon ecl-icon--xl ecl-social-media-follow__icon-hover\\"
        >
          <use xlink:href=\\"example#twitter_hover\\" /></svg
        ><span class=\\"ecl-link__label\\">Twitter</span></a
      >
    </li>
    <li class=\\"ecl-social-media-follow__item\\">
      <a
        href=\\"/example\\"
        class=\\"ecl-social-media-follow__link ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before\\"
        ><svg
          focusable=\\"false\\"
          aria-hidden=\\"true\\"
          class=\\"ecl-link__icon ecl-icon ecl-icon--xl ecl-social-media-follow__icon\\"
        >
          <use xlink:href=\\"example#facebook\\" /></svg
        ><svg
          focusable=\\"false\\"
          aria-hidden=\\"true\\"
          class=\\"ecl-link__icon ecl-icon ecl-icon--xl ecl-social-media-follow__icon-hover\\"
        >
          <use xlink:href=\\"example#facebook_hover\\" /></svg
        ><span class=\\"ecl-link__label\\">Facebook</span></a
      >
    </li>
    <li class=\\"ecl-social-media-follow__item\\">
      <a
        href=\\"/example\\"
        class=\\"ecl-social-media-follow__link ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before\\"
        ><svg
          focusable=\\"false\\"
          aria-hidden=\\"true\\"
          class=\\"ecl-link__icon ecl-icon ecl-icon--xl ecl-social-media-follow__icon\\"
        >
          <use xlink:href=\\"example#gplus\\" /></svg
        ><svg
          focusable=\\"false\\"
          aria-hidden=\\"true\\"
          class=\\"ecl-link__icon ecl-icon ecl-icon--xl ecl-social-media-follow__icon-hover\\"
        >
          <use xlink:href=\\"example#gplus_hover\\" /></svg
        ><span class=\\"ecl-link__label\\">Google+</span></a
      >
    </li>
    <li class=\\"ecl-social-media-follow__item\\">
      <a
        href=\\"/example\\"
        class=\\"ecl-social-media-follow__link ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before\\"
        ><svg
          focusable=\\"false\\"
          aria-hidden=\\"true\\"
          class=\\"ecl-link__icon ecl-icon ecl-icon--xl ecl-social-media-follow__icon\\"
        >
          <use xlink:href=\\"example#linkedin\\" /></svg
        ><svg
          focusable=\\"false\\"
          aria-hidden=\\"true\\"
          class=\\"ecl-link__icon ecl-icon ecl-icon--xl ecl-social-media-follow__icon-hover\\"
        >
          <use xlink:href=\\"example#linkedin_hover\\" /></svg
        ><span class=\\"ecl-link__label\\">Linkedin</span></a
      >
    </li>
    <li class=\\"ecl-social-media-follow__item\\">
      <a
        href=\\"/example\\"
        class=\\"ecl-social-media-follow__link ecl-link ecl-link--standalone\\"
        ><span class=\\"ecl-link__label\\">Other social networks</span></a
      >
    </li>
  </ul>
</div>
"
`;
